# HG changeset patch
# User John D
# Date 1483627036 18000
#      Thu Jan 05 09:37:16 2017 -0500
# Branch stable
# Node ID d591b38e34106cc153edd165a8d445e053640851
# Parent  14f1738a56ae5155b5a3ebee8b1972276fa0d7e2
Use GetModuleFileName for getting octave path in windows (bug #48671)

* src/shared-fcns.h (w32_get_octave_home): Use GetModuleFileName instead
of Module32First/Next scan.

diff --git a/src/shared-fcns.h b/src/shared-fcns.h
--- a/src/shared-fcns.h
+++ b/src/shared-fcns.h
@@ -35,49 +35,26 @@ along with Octave; see the file COPYING.
 
 static std::string
 w32_get_octave_home (void)
 {
   std::string retval;
 
   std::string bin_dir;
 
-  HANDLE h = CreateToolhelp32Snapshot (TH32CS_SNAPMODULE
-#if defined (TH32CS_SNAPMODULE32)
-                                       | TH32CS_SNAPMODULE32
-#endif
-                                       , 0);
-
-  if (h != INVALID_HANDLE_VALUE)
+  char namebuf[MAX_PATH+1];
+  if (GetModuleFileName (GetModuleHandle (NULL), namebuf, MAX_PATH))
     {
-      MODULEENTRY32 mod_info;
-
-      ZeroMemory (&mod_info, sizeof (mod_info));
-      mod_info.dwSize = sizeof (mod_info);
+      namebuf[MAX_PATH] = '\0';
 
-      if (Module32First (h, &mod_info))
-        {
-          do
-            {
-              std::string mod_name (mod_info.szModule);
-
-              if (mod_name.find ("octave") != std::string::npos)
-                {
-                  bin_dir = mod_info.szExePath;
+      std::string exe_name = namebuf; 
+      size_t pos = exe_name.rfind ("\\");
 
-                  if (bin_dir[bin_dir.length () - 1] != '\\')
-                    bin_dir.append (1, '\\');
-
-                  break;
-                }
-            }
-          while (Module32Next (h, &mod_info));
-        }
-
-      CloseHandle (h);
+      if (pos != std::string::npos)
+        bin_dir = exe_name.substr (0, pos + 1);
     }
 
   if (! bin_dir.empty ())
     {
       size_t pos = bin_dir.rfind ("\\bin\\");
 
       if (pos != std::string::npos)
         retval = bin_dir.substr (0, pos);
